added the show commands in the docs (#2951)
hashicorp/web-unified-docs: content/terraform/v1.16.x (beta)/docs/cli/commands/stacks/configuration/show.mdx

---
page_title: terraform stacks configuration show command
description: The terraform stacks configuration show command shows details of a configuration
# START AUTO GENERATED METADATA, DO NOT EDIT
created_at: 2026-08-03T17:10:24.000Z
last_modified: 2026-08-03T17:10:24.000Z
# END AUTO GENERATED METADATA
---

# `terraform stacks configuration show` reference

Use the `terraform stacks configuration show` command to show details of a configuration.

## Usage

```shell-session
$ terraform stacks <global-stacks-flags> configuration show <options>
```

## Description

The `terraform stacks configuration show` command retrieves relevant information for a given configuration, including its deployment groups, ingress attributes, and diagnostic help . 

> You can identify the target configuration either by providing a configuration ID directly, or by specifying the organization name, project name, and Stack name combination. When name-based targeting is used, the latest configuration for that stack is shown.

Reviewer: ```suggestion
You can target a configuration by providing either its ID, or by specifying all of the following values associated with the configuration:

- organization name 
- project name
- Stack name 

When specifying names, Terraform shows the latest configuration for the Stack.
```
 A little easier to read?

## Options

- `-organization-name`: The name of the organization to target.

>     - Optional when using `-stack-id`, required when using name-based targeting.

Reviewer: ```suggestion
    - Required when omitting the [`-configuration-id`](#configuration-id) flag.
```
Right? Not `stack-id`? 
The pattern we've implemented is to only state when the flag is required.

    - String data type.

>     - Overrides the ENV VAR `TF_STACKS_ORGANIZATION_NAME` if provided.

Reviewer: ```suggestion
    - Overrides the `TF_STACKS_ORGANIZATION_NAME` environment variable.
```

- `-project-name`: The name of the project to target.

>     - Optional when using `-stack-id`, required when using name-based targeting.
>     - String data type.
>     - Overrides the ENV VAR `TF_STACKS_PROJECT_NAME` if provided.

Reviewer: ```suggestion
    - Required when omitting the [`-configuration-id`](#configuration-id) flag.
    - String data type.
    - Overrides the `TF_STACKS_PROJECT_NAME` environment variable.
```

- `-stack-name`: The name of the stack to target.

>     - Optional when using `-stack-id`, required when using name-based targeting.
>     - String data type.
>     - Overrides the ENV VAR `TF_STACKS_STACK_NAME` if provided.

Reviewer: ```suggestion
    - Required when omitting the [`-configuration-id`](#configuration-id) flag.
    - String data type.
    - Overrides `TF_STACKS_STACK_NAME` environment variable.
```

- `-configuration-id`: The ID of the configuration to show.

>     - Optional (but required when not using name-based targeting).
>     - String data type.
>     - Has precedence over the latest configuration of the stack provided in named args.

Reviewer: ```suggestion
    - Required when omitting the [`-stack-name`](#stack-name), [`-project-name`](#project-name), and [`-organization-name`](#organization-name) flags.
    - String data type.
    - When you provide both `-configuration-id` and the [`-stack-name`](#stack-name), [`-project-name`](#project-name), and [`-organization-name`](#organization-name) flags, Terraform shows the specific configuration per the `-configuration-id` instead of the latest Stack configuration returned by the other arguments.
```
Clearer?

> - `-json`: Output results in JSON format.
>     - Optional.
>     - Boolean flag.
>     - Default is false (human-readable output).

Reviewer: ```suggestion
- `-json`: Formats the output in JSON format. When set to `false`, Terraform formats the output as human-readable text.
    - Optional.
    - Boolean flag.
    - Default is `false`.
```
